### PR TITLE
E2E setup

### PR DESCRIPTION
**Pull request by lorismoulin**
## Summary

Introduces a full E2E testing infrastructure using polkadot-api (papi) and sets up the MEV Shield test suite as the first consumer.

### E2E infrastructure (`e2e/`)

- Set up a pnpm workspace under `e2e/` with a `shared` utilities package and per-suite test packages.
- `e2e/shared` provides helpers for spawning multi-node networks, generating/patching chain specs, managing dev signers via polkadot-api, and a custom vitest sequencer for ordered test execution.
- Added `setup_env.sh` to check/install prerequisites (nvm, node, pnpm, jq, yq).
- Added `bootstrap_types.sh` to build the node binary and generate papi type descriptors without requiring dependencies to be installed first.
- Added `bootstrap_package.sh` to scaffold new E2E test packages using `jq`/`yq`.
- All temp files stored under `/tmp/subtensor-e2e/<suite>/` for easy bulk cleanup.

### MEV Shield E2E suite (`e2e/shield`)

- Migrated from dedot to polkadot-api.
- Tests cover: key rotation, encrypted transaction happy path, failed inner transactions, malformed ciphertext rejection, wrong key hash, stale key rejection, network scaling (3 to 6 nodes), and edge cases.
- Storage queries for rotating keys use `{ at: "best" }` instead of the default `"finalized"` to avoid stale key issues caused by GRANDPA finalization lag.

### CODE REVIEW

File: e2e/shield/setup.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+};
+
+const NODE_CONFIGS: NodeConfig[] = [
+  { name: "one", port: 30333, rpcPort: 9944, basePath: "/tmp/e2e-shield-one", validator: true },
+  { name: "two", port: 30334, rpcPort: 9945, basePath: "/tmp/e2e-shield-two", validator: true },
+  {
+    name: "three",
+    port: 30335,
+    rpcPort: 9946,
+    basePath: "/tmp/e2e-shield-three",
```
Comment (shamil-gadelshin) [inline e2e/shield/setup.ts]: IIRC we discussed a subfolder to simplify the the local cleaning script.

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]
The workflow misses the papi descriptors step:

https://github.com/opentensor/subtensor/actions/runs/22347000232/job/64665829129

File: e2e/shield/setup.ts (hunk 1 above)
Comment (lorismoulin) [reply e2e/shield/setup.ts]: Done, everything is under /tmp/subtensor-e2e that way we can just rm -rf

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -1,4 +1,146 @@
...
+  # Build the node binary in both profiles and share as artifacts.
+  build:
+    runs-on: [self-hosted, type-ccx33]
+    timeout-minutes: 60
+    strategy:
+      matrix:
+        include:
+          - profile: release
+            output_dir: release
+          - profile: dev
```
Comment (shamil-gadelshin) [inline .github/workflows/e2e.yml]: Why did you add the debug mode?

**Review by shamil-gadelshin — DISMISSED** [review]
Production build is much longer than release. I'll probably downgrade the fast build back to release later. Looks good!

**Comment by l0r1s** [comment]
> Production build is much longer than release. I'll probably downgrade the fast build back to release later. Looks good!

alright, just updated to release build

**Review by shamil-gadelshin — DISMISSED** [review]

**Review by sam0x17 — APPROVED** [review]

**Review by JohnReedV — APPROVED** [review]
